vslib/SwitchStateBaseMACsec.cpp

```diff
@@ -708,7 +704,15 @@ sai_status_t SwitchStateBase::loadMACsecAttrFromMACsecSA(
 
         // The Linux kernel directly uses ssci to XOR with the salt that is network order,
         // So, this conversion is useful to convert SSCI from the host order to network order.
-        macsecAttr.m_ssci = htonl(attr->value.u32);
+        //
+        // Starting with Debian Bookworm (iproute2 6.1), ssci is interpreted as a hex string,
+        // so this needs to convert the ssci integer to a hex string and account for encoding.
+        auto ssci = attr->value.u32;
```

Comment:
why you need extra value if this is used only once ?

Reply:
Good point. Updated the PR so that the extra variable isn't created, and updated the comment.